sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolvePartitionSpec.scala

```diff
@@ -49,6 +49,10 @@ object ResolvePartitionSpec extends Rule[LogicalPlan] {
           }
         case _ => command
       }
+    case s @ ShowTableExtended(_, _, partitionSpec @ Some(UnresolvedPartitionSpec(_, _)), _) =>
+      val extractPartitionSpec = new ExtractPartitionSpec(
+        partitionSpec.get.asInstanceOf[UnresolvedPartitionSpec])
+      s.copy(partitionSpec = Some(extractPartitionSpec))
   }
 
```

Comment:
In this scenario, UnresolvedPartitionspec cannot be resolved to ResolvedPartitionspec, because the table is non-deterministic at this time.